.github/workflows/tests.yml

```diff
@@ -28,25 +28,17 @@ jobs:
         with:
           python-version: "3.11"
 
-      - name: Set PY variable
-        run: echo "PY=$(python -VV | sha256sum | cut -d' ' -f1)" >> $GITHUB_ENV
-
-      - uses: actions/cache@v3
-        with:
-          path: ~/.cache/pre-commit
-          key: pre-commit|${{ env.PY }}|${{ hashFiles('.pre-commit-config.yaml') }}
-
       - name: Install pre-commit
         run: |
-          pip install 'pre-commit>=3.2.0,<4.0.0'
+          pip install 'pre-commit==3.3.2'
```

Comment:
Why the exact pin now?

Reply:
Well, I figured we might as well use stricter constraints to make sure everything is reproducible... Although pinning the patch version is probably not a great idea.

Reply:
I think usually we pin to the minor at most, as to avoid dependency conflicts for users in their venvs (and it's rare that there are breaking changes in between minors).

Reply:
This is just for the tests though right, so it's not that big a deal?